src/CMakeLists.txt

```diff
@@ -276,18 +276,20 @@ add_subdirectory(json_spirit)
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/xxHash")
 include_directories(SYSTEM "${CMAKE_SOURCE_DIR}/src/rapidjson/include")
 
+find_package(fmt 4.0.0 QUIET)
+if(NOT fmt_FOUND)
+  message(STATUS "Could not find fmt, will build it")
+  add_subdirectory(fmt)
+elseif(fmt_VERSION VERSION_GREATER_EQUAL 5.0.0)
+  message(WARNING "Could NOT find fmt: "
+    "Found unsuitable version \"${fmt_VERSION}\", "
+    "but required is at most \"5.0.0\" (found ${fmt_INCLUDE_DIR}). "
+    "Will build it")
+  add_subdirectory(fmt)
+  include_directories("{CMAKE_SOURCE_DIR}/src/fmt/")
```

Comment:
this `include_directories()` and the one in ../CMakeLists.txt shouldn't be necessary, because fmt attaches these to the target with `target_include_directories(fmt PUBLIC ...)`. this way you only get the include flags for the targets that need it. otherwise the cmake bits look good :+1: